lib/pmdtester/builders/diff_builder.rb

```diff
@@ -7,25 +7,29 @@ class DiffBuilder
     # http://pmd.sourceforge.net/report_2_0_0.xsd
 
     def build(base_report, patch_report)
-      base_doc = Nokogiri::XML(File.read(base_report))
-      patch_doc = Nokogiri::XML(File.read(patch_report))
+      base_doc = Nokogiri::XML(File.read(base_report)).remove_namespaces!
+      patch_doc = Nokogiri::XML(File.read(patch_report)).remove_namespaces!
 
-      violations_diffs = build_violation_diffs(base_doc, patch_doc)
+      violation_diffs = build_violation_diffs(base_doc, patch_doc)
       error_diffs = build_error_diffs(base_doc, patch_doc)
 
-      [violations_diffs, error_diffs]
+      [violation_diffs, error_diffs]
     end
 
     def build_diffs(base_hash, patch_hash)
-      base_hash.merge(patch_hash) do |key, base_value, patch_value|
-        base_value | patch_value - base_value & patch_value
+      diffs = base_hash.merge(patch_hash) do |key, base_value, patch_value|
+        (base_value | patch_value) - (base_value & patch_value)
+      end
+
+      diffs.delete_if do |key, value|
+        value.empty?
       end
```

Comment:
Hm... is it for sure, that this function always returns "diffs"? E.g. does it really return diffs, even if the computed hash contained empty values and thus "value.empty?" has been called? I don't see exactly, what the last executed statement of this function is and personally I would add an explicit "return diffs" as the last statement (or just "diffs") to make it clear, that this is the result of the function...

Reply:
@adangel 
> Hm... is it for sure, that this function always returns "diffs"?

Yes, the function alwats returns "diffs". Actually, `value.empty?` is is not accessible outside each's block, so we don't have to worry about the return value